WooCommerce/Classes/Authentication/Epilogue/StorePickerCoordinator.swift

```diff
@@ -51,16 +54,19 @@ extension StorePickerCoordinator: StorePickerViewControllerDelegate {
             return
         }
 
-        logOutOfCurrentStore(onCompletion: onCompletion)
+        switchStoreUseCase.logOutOfCurrentStore(onCompletion: onCompletion)
     }
 
     func didSelectStore(with storeID: Int64) {
         guard storeID != ServiceLocator.stores.sessionManager.defaultStoreID else {
             return
         }
 
-        finalizeStoreSelection(storeID)
-        presentStoreSwitchedNotice()
+        switchStoreUseCase.finalizeStoreSelection(storeID)
```

Comment:
I think we can make both the `logOutOfCurrentStore` and `finalizeStoreSelection` `private` by:

1. Adding this in `logOutOfCurrentStore`:

   ```swift
        guard stores.sessionManager.defaultStoreID != nil else {
            return onCompletion()
        }
    ```
2. Calling `updateDefaultStore` before the completion block call:

    ```swift
        group.notify(queue: .main) {
            finalizeStoreSelection(storeID: nil)
            onCompletion()
        }
    ```
3. And then in this `StorePickerCoordinator`, we'll only call:

    ```
    switchStoreUseCase.switchStore(with: storeID) { .... }
    ```

This requires some additional refactoring for `stores.updateDefaultStore()` and is a routine change because we will not be doing anything in `willSelectStore` anymore. That worries me because there's probably a reason why `StorePickerViewControllerDelegate` has two callbacks. 😅 So it's better if we do that only when we have some decent coverage. Another problem for another day, I guess!

Reply:
I did all the changes and I tried it, and it seems that it works fine. Can you take a look at it?